### PR TITLE
Improved and fixed the report incident command handler

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Hopefully this works. I have no idea. Let's see what travis makes of it all.

### CODE REVIEW

File: app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php [hunk 1]
```diff
@@ -29,24 +29,29 @@ class ReportIncidentCommandHandler
...
+        ];
+
+        // Link with the component.
+        if ($command->component_id) {
+            $data['component_id'] = $command->component_id;
+        }
 
         // The incident occurred at a different time.
         if ($command->incident_date) {
             $incidentDate = Date::createFromFormat('d/m/Y H:i', $command->incident_date, config('cachet.timezone'))->setTimezone(Config::get('app.timezone'));
```
Comment (GrahamCampbell) [inline app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php]: also, we shouldn't ever be using the config helper here

Comment (GrahamCampbell) [inline app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php]: all "foundation" global functions should be banned unless they're in controllers

Comment (GrahamCampbell) [inline app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php]: our handler classes should be framework agnostic

Comment (jbrooksuk) [inline app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php]: So pass it through to the command?

Comment (GrahamCampbell) [inline app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php]: no

Comment (GrahamCampbell) [inline app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php]: we need to bind the handler to the ioc and inject the config value at that point

Comment (GrahamCampbell) [inline app/Handlers/Commands/Incident/ReportIncidentCommandHandler.php]: we shouldn't abuse commands to pass config through

**Comment by GrahamCampbell** [comment]
Travis passing. Whoop whoop.

**Comment by jbrooksuk** [comment]
:tada:

**Pull request merged**